### ISSUE TITLE
Support for Windows containers in Radix

### ISSUE THREAD

**Issue by magnus-longva-bouvet** [equinor/radix-platform#391]
**Is your feature request related to a problem? Please describe.**
The Oilmod1 team is migrating applications to Omnia Classic. We are anticipating that Radix will be available in Omnia Classic some time during Q1 or early Q2 in 2021. We have three applications with tight couplings to Windows OS. One of them, `Veson Imos Integration`, is a third party Windows binary for which we don't have the source code. The second application, `RAD`, is written in Java and uses the [WAFFLE](https://github.com/Waffle/waffle) library for Kerberos client authentication (I know there might be other obstacles than OS to make that one fly in Radix). The third application, `eTradeHub`, relies on a third party software called `Evo HTML to PDF Converter` ([link](https://www.evopdf.com/)), and this software only runs on Windows.

**Describe the solution you'd like**
I'd like Radix to support Windows containers. If there's a situation where either Nano Server or Server Core must be prioritized, I would prefer Server Core.

**Describe alternatives you've considered**
For `RAD`:
- Rewrite the Kerberos authentication mechanism to use some other, cross-platform library.
- Rewrite to use something else than Kerberos authentication.
- Use ACI Windows containers. Windows containers deployet to vnets will become a preview feature some time in February 2021.
- Run the app on a virtual machine, and deal with all the overhead that comes with it.

For `Veson Imos Integration`:
- Run the application on a Linux container with `wine` emulation. Might be difficult, because the application's logic has tight couplings to the concept of Windows services.
- Write our own implementation of the app.
- Use ACI Windows containers. Windows containers deployet to vnets will become a limited preview feature some time in April 2021.
- Run the app on a virtual machine, and deal with all the overhead that comes with it.

For `eTradeHub`:
- Run the `Evo HTML to PDF Converter` application on a Linux container with `wine` emulation. Have tried, but have not succeed.
- Find some other HTML to PDF conversion software which runs natively on Linux, and rewrite our client integrations.
- Use ACI Windows containers. Windows containers deployet to vnets will become a limited preview feature some time in April 2021.
- Run the app on a virtual machine, and deal with all the overhead that comes with it.

**Comment by magnus-longva-bouvet** [comment]
Solum is doing active development on AKS in Omnia Classic, so this issue is becoming pertinent again.
https://github.com/equinor/Solum/issues?q=is%3Aissue+label%3AAKS+

**Comment by emirgens** [comment]
We have performed a PoC using Windows Containers in Radix, however there are a couple of remaining issues which needs to be closed before we can continue implementing this feature.
Currently the AKS version does not support Calico Network isolation for Windows nodes
Proper way of configuring this in Radix is not available

Roadmap for a AKS version (1.22.x) supporting Calico is on our agenda.

**Comment by cbuv** [comment]
any updates on this? @emirgens

**Comment by oyron** [comment]
We looked into this about a year ago, then put it on hold when we hit technical obstacles. These issues might have been solved with newer versions of AKS, but we believe that supporting Windows containers will require a lot of work, and it is not given that we will be able to make it work smoothly.
With the support of running newer versions of .NET on Linux, the need for Windows containers seems to be reduced. Kerberos support could be another reason for teams wanting Win containers, but it is unlikely that Radix will ever be set up in Omnia Classic / with on-prem network connectivity.

Unless there is a considerable demand for Windows containers, the work on Windows containers will not be resumed. Hopefully you'll find other ways to solve your issues @cbuv .
